src/fides/api/models/sql_models.py

```diff
@@ -295,15 +295,43 @@ class Dataset(Base, FidesBase):
     collections = Column(JSON)
     fides_meta = Column(JSON)
 
+    @classmethod
+    def create(
+        cls: Type["Dataset"],
+        db: Session,
+        *,
+        data: dict[str, Any],
+        check_name: bool = True,
+    ) -> "Dataset":
+        """Override create to check for existing datasets with the same fides_key or name."""
+        # Check if dataset with same fides_key already exists
+        if "fides_key" in data:
+            existing_by_key = (
+                db.query(cls).filter(cls.fides_key == data["fides_key"]).first()
+            )
+            if existing_by_key:
+                raise KeyOrNameAlreadyExists(
+                    f'Dataset with fides_key "{data["fides_key"]}" already exists.'
+                )
```

Comment:
We can't use the default "key or name" check because we pass in a `fides_key` instead of a `key`